src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,69 @@
 package core.basesyntax;
 
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
+    private static final String SUPPLY = "supply";
+    private static final String BUY = "buy";
+    private static final String RESULT = "result";
+    private static final String WORD_SPLIT_REGEX = "\\W+";
+
     public void getStatistic(String fromFileName, String toFileName) {
+        File inputFile = new File(fromFileName);
+        String[] words = readFromFile(inputFile).split(WORD_SPLIT_REGEX);
+
+        int supplySum = 0;
+        int buySum = 0;
+
+        for (int i = 0; i < words.length; i += 2) {
+            if (words[i].equals(SUPPLY)) {
+                supplySum += Integer.parseInt(words[i + 1]);
+            } else if (words[i].equals(BUY)) {
+                buySum += Integer.parseInt(words[i + 1]);
+            }
+        }
+
+        String report = generateReport(supplySum, buySum);
+        writeToFile(toFileName, report);
+    }
+
+    private String generateReport(int supplySum, int buySum) {
+        int resultSum = supplySum - buySum;
+        StringBuilder stringBuilder = new StringBuilder();
+        stringBuilder.append(SUPPLY).append(",").append(supplySum).append(System.lineSeparator())
```

Comment:
All magic numbers, strings, and symbols in your code should be constants.
[this article](https://mate-academy.github.io/style-guides/java/java.html#s5.2.4-constant-names)
```suggestion
        stringBuilder.append(SUPPLY).append(SEPARATOR).append(supplySum).append(System.lineSeparator())
```